src/parsing/binast-serialize-visitor.h

```diff
@@ -549,6 +558,18 @@ inline void BinAstSerializeVisitor::VisitFunctionLiteral(FunctionLiteral* functi
   for (Statement* statement : *function_literal->body()) {
     VisitNode(statement);
   }
+
+  // Calculate length and insert at length_index
+  auto length = byte_data_.size() - start;
+  for (size_t i = 0; i < sizeof(uint32_t) / sizeof(uint8_t); ++i) {
```

Comment:
Iiiinteresting...I wonder if there's a way we could de-dupe this with the SerializeUint32 logic...maybe we could provide an offset override so it would fill in with it's usual logic?

Reply:
yeah interesting idea! i was considering doing something like that but wasn't sure i had a great sense of what other situations we'd use it for? will give it a try if you think we'll find it useful elsewhere

Reply:
i suppose we'll want to do something like this for function literals altogether when we move them to the end of the serialized data?

Reply:
It just seems a little sad to let this low level bit-shifty stuff escape from within the Serialize<PrimitiveType> methods.